pennylane/dla/lie_closure.py

```diff
@@ -0,0 +1,258 @@
+            PauliSentence({
+                PauliWord({0:"X", 1:"X"}) : 1.,
+                PauliWord({0:"Y", 1:"Y"}) : 1.
+            }),
+            PauliSentence({
+                PauliWord({0:"X", 1:"X"}) : 1.,
+            }),
+            PauliSentence({
+                PauliWord({0:"Y", 1:"Y"}) : 1.,
+            }),
+        ]
+
+        vspace = PauliVSpace(ops)
+
+    It automatically detects that the third operator is linearly dependent on the former two, so it does not add the third operator to the basis.
+
+    >>> vspace.basis
+    [1.0 * X(0) @ X(1)
+     + 1.0 * Y(0) @ Y(1),
+     1.0 * X(0) @ X(1)]
+
+    We can also retrospectively add operators.
+
+    >>> vspace.add(PauliWord({0:"X"}))
+    [1.0 * X(0) @ X(1)
+     + 1.0 * Y(0) @ Y(1),
+     1.0 * X(0) @ X(1),
+     1.0 * X(0)]
+
+    Again, checks of linear independence are always performed. So in the following example no operator is added.
+
+    >>> vspace.add(PauliWord({0:"Y", 1:"Y"}))
+    [1.0 * X(0) @ X(1)
+     + 1.0 * Y(0) @ Y(1),
+     1.0 * X(0) @ X(1),
+     1.0 * X(0)]
+    """
+
+    def __init__(self, generators):
+
+        # Get all Pauli words that are present in at least one Pauli sentence
+        all_pws = list(reduce(set.__or__, [set(ps.keys()) for ps in generators]))
+        num_pw = len(all_pws)
+        # Create a dictionary mapping from PauliWord to row index
+        self._pw_to_idx = {pw: i for i, pw in enumerate(all_pws)}
+
+        # Initialize PauliVSpace properties trivially
+        self._basis = []
+        rank = 0
+
+        self._M = np.zeros((num_pw, rank), dtype=float)
+        self._rank = rank
+        self._num_pw = num_pw
+
+        # Add all generators that are linearly independent
+        self.add(generators)
+
+    @property
+    def basis(self):
+        """List of basis operators of PauliVSpace"""
+        return self._basis
+
+    def __len__(self):
+        return len(self.basis)
+
+    def add(self, other):
+        """Adding a list of PauliSentences if they are linearly independent"""
+        if isinstance(other, (PauliWord, PauliSentence, Operator)):
+            other = [other]
+
+        other = [
+            qml.pauli.pauli_sentence(op) if not isinstance(op, PauliSentence) else op
+            for op in other
+        ]
+
+        for ps in other:
+            # TODO: Potential speed-up by computing the maximal linear independent set for all current basis vectors + other, essentially algorithm1 in https://arxiv.org/abs/1012.5256
+            self._M, self._pw_to_idx, self._rank, self._num_pw, is_independent = (
+                self._check_independence(self._M, ps, self._pw_to_idx, self._rank, self._num_pw)
+            )
+            if is_independent:
+                self._basis.append(ps)
+        return self._basis
+
+    def is_independent(self, pauli_sentence, tol=1e-15):
+        r"""Check if the :class:`~PauliSentence` ``pauli_sentence`` is linearly independent with all vectors in ``PauliVSpace``.
```

Comment:
"with all the vectors" could rather easily be misread as "with each of the vectors", I think?
```suggestion
        r"""Check if the :class:`~PauliSentence` ``pauli_sentence`` is linearly independent of the basis of ``PauliVSpace``.
```